### DISCUSSION TITLE
CSS Custom Theming?

### DISCUSSION THREAD

**Discussion by PowershellScripter** [dani-garcia/vaultwarden#1954] [Q&A] [answered]
Is there any ways to use a custom .css file in the docker container of this image?

I modified the current one and overwrote it with mine and restarted the container but no luck getting it to take.

Is there something special i need to do or is it even possible?

**Comment by RealOrangeOne** [comment]
It should work in theory, but it's definitely not officially supported, and will almost certainly break between updates.

If you're modifying the file correctly, it should work. That implies it's either the wrong file, or it's not applying. But again that aligns with the idea it's probably a bad idea anyway.

**Reply by PowershellScripter** [reply]
In the docker container path, I can 'cat' the file and see my modified version, but in the browser under sources, it does not show the changed file. I understand that it will break between updates which is why I keep the modified code elsewhere to re-apply later on. I add my css mods to the bottom of the file so that the only changes modifying are the elements I chose to modify. In Stylus, it applies with no issues, but a direct swap out does nothing.

**Comment by BlackDex** [comment, marked as answer]
Best thing to do is add some bash script to the /etc/vaultwarden.d folder which appends some content to /web-vault/app/main.*.css.
So something like.
```bash
#!/bin/bash
cat /data/my-custom.css >> /web-vault/app/main.*.css
```

Or

```bash
#!/bin/bash
echo -e '\nbody { color: green; }\n' >> /web-vault/app/main.*.css
```

**Reply by PowershellScripter** [reply]
/etc/vaultwarden.d folder doesnt exist inside the docker container

**Reply by jjlin** [reply]
See https://github.com/dani-garcia/vaultwarden/wiki/Starting-a-Container#customizing-container-startup

**Reply by PowershellScripter** [reply]
So I tried both bin bash examples, mapping my volume to the /etc/vaultwarden.d directory and made my .sh executable. Upon restarting the container, neither of bash examples worked.

Update** I can go into the console and see that the file in /web-vault/app updated, but the portal does not reflect these changes.

Update again**
Nvm, looks like my browser is being stupid and caching the page. Incognito shows the changes. Thx for all your help :)

**Reply by BlackDex** [reply]
Did you cleared the browser cache? Or pressed ctrl+f5 ?

**Reply by PowershellScripter** [reply]
> Did you cleared the browser cache? Or pressed ctrl+f5 ?

Yea I fixed it! Thank you so much for your help! This makes it far easier to customize it in the future as I can keep a local customized css file and I created a simple bash script that looks for the fullname of the original css file in the directory you specific and 'cats' my file to that that one. And with the volume path being mounted, it survives updates with no issue (already tested with recreate of container). And if the css ever changes when I update, I just rename my .sh file to .sh.old until I correct the css elements, then just rename it back and voila. Thank you again, really!
